### PR TITLE
[14.0][MIG] account_move_line_purchase_info

### PR DESCRIPTION
**Pull request by ps-tubtim**
- Add `purchase_line_id`, `purchase_order_id` on form view of `account.move.line`

Selection_023

### CODE REVIEW

**Review by AaronHForgeFlow — APPROVED** [review]
Functional test on runbot. :+1:

File: account_move_line_purchase_info/models/account_move.py [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
-
-from odoo import fields, models
-
-
-class AccountMoveLine(models.Model):
-
-    _inherit = "account.move.line"
-
-    purchase_id = fields.Many2one(
```
Comment (LoisRForgeFlow) [inline account_move_line_purchase_info/models/account_move.py]: I think we should add a migration script to move this to the new standard field `purchase_order_id`.

Comment (LoisRForgeFlow) [reply account_move_line_purchase_info/models/account_move.py]: I update my comment: the field purchase_order_id is not stored, so therefore the filters added in this module won't work if we don't make it stored again.

Comment (AaronHForgeFlow) [reply account_move_line_purchase_info/models/account_move.py]: There are two options here:
- Move purchase_id field to the new standard field purchase_order_id, and make it stored: there is a risk on dependent fields. Dependent fields may not me computed correctly. A migration script will be needed to rename the fields.
- Keep separate fields: duplicated fields is a dirty option but safer and no migration script needed. But dirty

After choosing one option of those, can you please update the roadmap in the readme with the details of what and why we chose that option?

**Comment by AaronHForgeFlow** [comment]
Could you also include https://github.com/OCA/account-financial-tools/pull/1219 ? Ty!

File: account_move_line_purchase_info/models/account_move.py (hunk 1 above)
Comment (ps-tubtim) [reply account_move_line_purchase_info/models/account_move.py]: @LoisRForgeFlow  @AaronHForgeFlow Thank you for your suggestions. I choose option

- > Move purchase_id field to the new standard field purchase_order_id, and make it stored: there is a risk on dependent fields. Dependent fields may not me computed correctly. A migration script will be needed to rename the fields.

Now I add a migration script, which you can review again.

File: account_move_line_purchase_info/migrations/13.0.1.1.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020 ForgeFlow (https://www.forgeflow.com)
```
Comment (MiquelRForgeFlow) [inline account_move_line_purchase_info/migrations/13.0.1.1.0/post-migration.py]: delete this file

File: account_move_line_purchase_info/migrations/14.0.1.0.0/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    move_line_table_name = "account_move_line"
+    ou_purchase_rename = openupgrade.get_legacy_name(old_purchase_column)
+
+    if openupgrade.column_exists(env.cr, move_line_table_name, ou_purchase_rename):
+        openupgrade.rename_columns(
+            env.cr,
+            {
+                move_line_table_name: [(ou_purchase_rename, new_purchase_column)],
+            },
+        )
```
Comment (MiquelRForgeFlow) [inline account_move_line_purchase_info/migrations/14.0.1.0.0/pre-migration.py]:
```suggestion

    if not openupgrade.column_exists(env.cr, 'account_move_line', new_purchase_column):
        openupgrade.rename_fields(env, [
    ('account.move.line', 'account_move_line', old_purchase_column, new_purchase_column),
])
```

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:  Functional test + code review, after applying the requested changes by @MiquelRForgeFlow  in the migration script

File: account_move_line_purchase_info/migrations/13.0.1.1.0/post-migration.py (hunk 2 above)
Comment (ps-tubtim) [reply account_move_line_purchase_info/migrations/13.0.1.1.0/post-migration.py]: I already deleted this file.

File: account_move_line_purchase_info/migrations/14.0.1.0.0/pre-migration.py (hunk 3 above)
Comment (ps-tubtim) [reply account_move_line_purchase_info/migrations/14.0.1.0.0/pre-migration.py]: @MiquelRForgeFlow  Thank you for your suggestion. I already update this file.

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**